### PR TITLE
How to use Go Fiber and Gorm Frameworks to run a Golang Application

### LINKED ISSUE #4478: [Languages]How to use Go Fiber and Gorm Frameworks to run a Golang Application
#### Proposed title of the article
Your title should be descriptive of the article/tutorial. Be Specific. 

How to use Go Fiber and Gorm Frameworks to run a Golang Application

Use keyword research to improve your article's SEO.

#### Proposed article introduction
*We're looking for the first 2-3 paragraphs of your article that appropriately summarize what your article will be about.*
We will use this writing snippet to help us assess the overall quality before approval so ensure it has been proofread, spellchecker, and grammar-checked.

If you have worked with Node.js, you must have used Express. In one or two projects. Express is the most popular Node.js web server framework. If you're building a back end using JavaScript, then chances are you're going to be using Express. Express is the industry standard for creating servers in Node.js. This helps you control and executing the code from a server. Thus facilitating an easier way to manipulate server-side data such as a database.

On the other side Go, popularly known as Golang, is a compiled language. It is compiled down into machine-level code that computers understand. This means it doesn't need to be interpreted one line at a time, and that's one of the reasons it's so fast compared to interpreted languages like python.

Go is a general-purpose language meaning that it can be used for many different things. You can use go to build web apps, microservices, cloud services, APIs, DevOps toolings, and any application you can think of. This means you will probably use Go to build a server-side-based application. Just like Node.js, you will need the right tooling to be able to handle your server-side code.

Go provide you with different frameworks that help you do such tasks. This includes the Go Fiber Framework. Go Fiber is an Express-inspired framework for Golang. Go Fiber is a web framework built on top of fast HTTP. Just like Express, Go Fiber can be used to handle operations such as routing/endpoints, middleware, server request, etc.

In this tutorial, we will learn more about Go fiber. We will use Go fiber with Gorm and an SQLite database to build a todo application. This will help you understand the concept of Go fiber in detail. This includes creating middleware, routing, etc. All using the Goland programming language

#### Key takeaways 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
Use this area to get your ideas down on the bulk of your article or tutorial.

- Why Go
- What is Go Fiber
- What is Gorm
- Set up a basic Go fiber server.
- Set up a todo application to understand Go fiber in details

#### Article quality
How is your article unique? Tell us what makes your approach different from similar articles that have published on the same topic?
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquote notes as you fill in the form before you submit.
We look forward to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Joakim-gakure**
How to use Go Fiber and Gorm Frameworks to run a Golang Application

### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4478

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✔️.

File: content/articles/how-to-use-go-fiber-and-gorm-frameworks-to-run-a-golang-application/index.md [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+        "error": "todo not found",
+    })
+}
+```
+
+#### Create a new todo
+
+`AddTodo()` function created a new todo and saves it to the database. Here we are only adding `Name`. The `Completed` is set to be `false` by default. We will set `Completed` as `true` when updating completed todos.
+
+This is where the `uuid` comes into play. We use a UUID instance to generate and inspect our data structure before inserting it into the database. It generates an immutable Universally Unique IDentifier([UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier)) random number. Thus ensuring each id is unique to one todo.
```
Comment (jumamiller) [inline content/articles/how-to-use-go-fiber-and-gorm-frameworks-to-run-a-golang-application/index.md]: Explain code snippets after writing that code. This is quite confusing.

**Review by jumamiller — APPROVED** [review]
@Joakim-gakure this is a great article!

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
👍 overall nice work - very detailed and straight forward tutorial 
 
Please review and suggestions/changes made by your peer reviewer and/or myself during a final review.

**Pull request merged**